CHANGELOG.md

```diff
@@ -24,6 +25,7 @@ Introduces biowaste produced by residential units and some quality of life impro
 - Fix a mistake that tallied food totals before population consumption which resulted in incorrect value displayed in the HUD
 - Fix Mine full-screen UI showing "no routes" when loading a game
 - Fix for phantom mine routes being drawn on the Mini Map if you quit to the main menu from an active game and start a new game
+- Fix for immediate colony death when landing colonists

```

Comment:
If this only happens in non-tagged releases, there's probably no real need to add a Change Log entry for it.

Reply:
I'm pretty sure this is a problem that affected v0.8.0 - population requirement was made in that changeset.

I'm noticing that my testing of new versions of the game is inadequate as I assume that the beginning parts of the game are already done and tested and so I focus on running from save games. A lot of the fixes I implemented that aren't tied to specific issues are because I opted to try running the game from the very beginning. Something I'm going to have to change moving forward.

Reply:
Sounds like we could use some automated tests to help ensure quality.

Admittedly though, I'm not too sure how to go about writing tests for some of these cases. Most of our test code is unit tests, which are most appropriate for library code. Once you start involving the GUI, or a full executable, things get a little more complicated.

Though I suppose certain core game mechanics could be tested in a similar manner to library code, using unit tests.

Reply:
That's where I'm stumbling too. How do you automate tests on an interactive program that displays output to the user visually?

Some of the tests could be automated by simulating specific parts of the code? And simulating input? All of the user input is handled through discrete functions that should be testable. Gonna have to be something to really think about since there are a lot of variables to consider here (and I don't just mean actual program variables)